src/CentrifugeRouter.sol

```diff
@@ -17,20 +19,23 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
     }
 
     // --- Deposit ---
+
     /// @inheritdoc ICentrifugeRouter
-    function requestDeposit(address vault, uint256 amount) external {
-        IERC7540(vault).requestDeposit(amount, msg.sender, msg.sender);
+    function requestDeposit(address vault, uint256 amount) public {
+        SafeTransferLib.safeTransferFrom(IERC7540(vault).asset(), msg.sender, address(this), amount);
+        IERC20(IERC7540(vault).asset()).approve(vault, amount);
+        IERC7540(vault).requestDeposit(amount, msg.sender, address(this));
```

Comment:
Why is this reverted to the old logic?

Reply:
Maybe I missed this conversation somewhere, but without these lines the user has to explicitly approve the vault themselves, which I thought was something we were trying to avoid with this contract since its another UX focused contract.